### PR TITLE
feat(cascader): support showSearch.onChange

### PR DESCRIPTION
**Pull request by GSBL**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

[详见issue](https://github.com/ant-design/ant-design/issues/27696#issuecomment-725300178)
### 💡 Background and solution

[详见issue](https://github.com/ant-design/ant-design/issues/27696#issuecomment-725300178)
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Cascader support `showSearch={{ onChange }}`. |
| 🇨🇳 Chinese | Cascader 支持 `showSearch={{ onChange }}`。  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/cascader/demo/custom-search.md](https://github.com/GSBL/ant-design/blob/feature-cascader-onSearch/components/cascader/demo/custom-search.md)
[View rendered components/cascader/index.en-US.md](https://github.com/GSBL/ant-design/blob/feature-cascader-onSearch/components/cascader/index.en-US.md)
[View rendered components/cascader/index.zh-CN.md](https://github.com/GSBL/ant-design/blob/feature-cascader-onSearch/components/cascader/index.zh-CN.md)

### CODE REVIEW

File: components/cascader/index.zh-CN.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/UdS8y8xyZ/Cascader.svg
 | value | 指定选中项 | string\[] \| number\[] | - |  |
 | onChange | 选择完成后的回调 | (value, selectedOptions) => void | - |  |
 | onPopupVisibleChange | 显示/隐藏浮层的回调 | (value) => void | - |  |
+| onSearch | 文本框值变化时的回调，需要设置`showSearch`为`true` | function(value: string) | - |  |
```
Comment (afc163) [inline components/cascader/index.zh-CN.md:52]: onSearch 放到 showSearch 里面去。

Comment (GSBL) [reply components/cascader/index.zh-CN.md:52]: ok

File: components/cascader/index.zh-CN.md [hunk 2]
```diff
@@ -62,6 +61,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/UdS8y8xyZ/Cascader.svg
 | matchInputWidth | 搜索结果列表是否与输入框同宽（[效果](https://github.com/ant-design/ant-design/issues/25779)） | boolean | true |  |
 | render | 用于渲染 filter 后的选项 | function(inputValue, path): ReactNode | - |  |
 | sort | 用于排序 filter 后的选项 | function(a, b, inputValue) | - |  |
+| onSearch | 文本框值变化时的回调 | function(value: string) | - |  |
```
Comment (afc163) [inline components/cascader/index.zh-CN.md:64]: 改成 onChange 是不是更准确一点。

Comment (GSBL) [reply components/cascader/index.zh-CN.md:64]: ok

File: components/cascader/index.en-US.md [hunk 3]
```diff
@@ -58,6 +58,7 @@ Cascade selection box.
 | matchInputWidth | Whether the width of list matches input, ([how it looks](https://github.com/ant-design/ant-design/issues/25779)) | boolean | true |  |
 | render | Used to render filtered options | function(inputValue, path): ReactNode | - |  |
 | sort | Used to sort filtered options | function(a, b, inputValue) | - |  |
+| onChange | A callback function, can be executed when the search input changes | function(value: string) | - |  |
```
Comment (hengkx) [inline components/cascader/index.en-US.md:61]: version  4.9.0

Comment (GSBL) [reply components/cascader/index.en-US.md:61]: 不懂什么意思

Comment (GSBL) [reply components/cascader/index.en-US.md:61]: > version 4.9.0

什么意思啊

**Review by afc163 — APPROVED** [review]

**Comment by zombieJ** [comment]
还是应该 onSearch，Select & TreeSelect 都是 `searchValue` & `onSearch` 做受控访问。

**Comment by hengkx** [comment]
还没发发版 还有 机会

**Comment by zombieJ** [comment]
嗯，我先 revert 掉。免得不小心被我发了。
@GSBL 帮忙再搞一下  `onSearch` 哈~~

**Comment by GSBL** [comment]
> 嗯，我先 revert 掉。免得不小心被我发了。
> @GSBL 帮忙再搞一下 `onSearch` 哈~~

@zombieJ 只用把onChange重命名onSearch, 还是需要重showSearch中提出来

**Comment by Kylinyu** [comment]
大佬们，我使用antd4，没看到这个选项用来自定义搜索，请问上线了么？

**Pull request merged**